### PR TITLE
Adding new interaction engine and graphic renderer to develop branch

### PR DESCRIPTION
**Pull request by Amarcolina**
This PR is the big one!  We add the new interaction engine, and the graphic renderer into the develop branch in preparation for the release next thursday!  All remaining changes will be merged directly into develop! 

To test:
 - [x] Make sure everything works.
 - [x] Make sure nothing breaks.

### CODE REVIEW

File: Assets/LeapMotionModules/GraphicRenderer/Testing/Scripts/GraphicRendererTestBase.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+ * Copyright (C) Leap Motion, Inc. 2011-2017.                                 *
+ * Leap Motion proprietary and  confidential.                                 *
+ *                                                                            *
+ * Use subject to the terms of the Leap Motion SDK Agreement available at     *
+ * https://developer.leapmotion.com/sdk_agreement, or another agreement       *
+ * between Leap Motion and you, your company or other organization.           *
+ ******************************************************************************/
+
+using System;
+using NUnit.Framework;
```
Comment (nickjbenson) [inline Assets/LeapMotionModules/GraphicRenderer/Testing/Scripts/GraphicRendererTestBase.cs]: The tests folder breaks Graphic Renderer example builds currently. I think we need to put the Testing scripts behind an Editor folder?

File: Assets/LeapMotionModules/GraphicRenderer/Resources/GraphicRenderer.cginc [hunk 2]
```diff
@@ -2,4 +2,4 @@
 
 // This line is auto-generated, do not modify!
 // Maximum graphic count can be changed in the preferences dialog.
-#define GRAPHIC_MAX 144
+#define GRAPHIC_MAX 48
```
Comment (jselstad) [inline Assets/LeapMotionModules/GraphicRenderer/Resources/GraphicRenderer.cginc]: Why was this lowered (and not a power of 2) @Amarcolina ?

Comment (nickjbenson) [inline Assets/LeapMotionModules/GraphicRenderer/Resources/GraphicRenderer.cginc]: It was a fix for DX9 that was breaking builds

Comment (Amarcolina) [inline Assets/LeapMotionModules/GraphicRenderer/Resources/GraphicRenderer.cginc]: Yeah, DX9 apparently has a very small amount of registers that can hold data :( Doesn't stop you from going much higher than this as long as you don't want to support dx9

**Pull request merged**